contracts/CumulativeMerkleDrop.sol

```diff
@@ -15,25 +13,39 @@ contract CumulativeMerkleDrop is Ownable, ICumulativeMerkleDrop {
     using MerkleProof for bytes32[];
 
     address public immutable override token;
+    address public rewardsHolder;
 
     bytes32 public override merkleRoot;
     mapping(address => uint256) public cumulativeClaimed;
+    struct Claim {
+        address account;
+        uint256 amount;
+        bytes32[] proof;
+    }
 
-    constructor(address token_) {
+    event RewardsHolderUpdated(address oldRewardsHolder, address newRewardsHolder);
+
+    constructor(address token_, address rewardsHolder_) {
         token = token_;
+        rewardsHolder = rewardsHolder_;
     }
```

Comment:
I know that is not necessary but would be great to add check for token contract here, something like `token.totalSupply > 0`. And that is `rewardsHolder`  is not zero address

Reply:
good idea